core/src/main/java/com/linecorp/armeria/server/auth/AuthService.java

```diff
@@ -110,22 +136,39 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
 
     private HttpResponse handleSuccess(HttpService delegate,
                                        @Nullable AuthSuccessHandler authorizerSuccessHandler,
-                                       ServiceRequestContext ctx, HttpRequest req)
+                                       ServiceRequestContext ctx, HttpRequest req,
+                                       long startNanos)
             throws Exception {
         final AuthSuccessHandler handler = authorizerSuccessHandler == null ? defaultSuccessHandler
                                                                             : authorizerSuccessHandler;
-        return handler.authSucceeded(delegate, ctx, req);
+        try {
+            return handler.authSucceeded(delegate, ctx, req);
+        } finally {
+            maybeRecordTimer(successTimer, startNanos);
```

Comment:
The timer will include the additional time it takes to fully complete the request, which doesn't address the original issue. Let me move this before calling the handler. 😉